### ISSUE TITLE
Misidentifying subclass of (python) list?

### ISSUE THREAD

**Issue by rjplevin** [JuliaPy/PyCall.jl#507]
I'm trying to wrap a package that defines a class that subtypes python's `list` and adds methods. When I create one of these, it appears to be misidentified as a `list`, so the additional methods are not available. 

The offending line seems to be conversions.jl:762:
```
        # only handle PyList for now
        return pyisinstance(o, @pyglobalobj :PyList_Type) ? Array : Union{}
```
Would it break things to check if `type(o) == list` rather than whether its an instance?

Example:
```
Main> py"class MyClass1(list):    # subclass of list
            def fn(self):
                print 'waldo was here'
       "

Main> py"class MyClass2():     # NOT a subclass of list
            def fn(self):
                print 'waldo was here'
       "

Main> c1 = py"MyClass1()"     # misidentified as a list => Array
0-element Array{Any,1}

Main> c2 = py"MyClass2()"
PyObject <__main__.MyClass2 instance at 0x1a48492758>

Main> py"$c1.fn()"
ERROR: PyError (ccall(@pysym(:PyEval_EvalCode), PyPtr, (PyPtr, PyPtr, PyPtr), o, globals, locals)) <
type 'exceptions.AttributeError'>
AttributeError("'list' object has no attribute 'fn'",)
  File "/Users/rjp/.julia/v0.6/PyCall/src/pyeval.jl", line 1, in <module>
    const Py_single_input = 256  # from Python.h

Stacktrace:
 [1] pyerr_check at /Users/rjp/.julia/v0.6/PyCall/src/exception.jl:60 [inlined]
 [2] pyerr_check at /Users/rjp/.julia/v0.6/PyCall/src/exception.jl:64 [inlined]
 [3] macro expansion at /Users/rjp/.julia/v0.6/PyCall/src/exception.jl:84 [inlined]
 [4] pyeval_(::String, ::PyCall.PyDict{String,PyCall.PyObject,false}, ::PyCall.PyDict{String,PyCall.
PyObject,false}, ::Int64, ::String) at /Users/rjp/.julia/v0.6/PyCall/src/pyeval.jl:23
 [5] eval(::Module, ::Any) at ./boot.jl:235

Main> py"$c2.fn()"
waldo was here
```

**Comment by oxinabox** [comment]
I think this would be a good change.
It is hitting me for Dicts right now. And more generally it has been bugging me for years.
The work around is to use `pycall(func, PyObject, args...)` which avoids converting the type.

But I think automatic type-conversion should only occur for exact types of Dict and List etc, and not there subtypes.
I think there should be a function that one can call if one does want to convert them.

eg so that for some `Foo[:get_special_list]()` which currently returns a `Vector` but should return a `PyObject` representing a subtype of a python list.
Then `jltype(Foo[:get_special_list]())`   would get a `Vector`
maybe be explict and one would call `convert(Vector,Foo[:get_special_list]())`

**Comment by oxinabox** [comment]
Oh apparently in the new PyCall (and I say new because I've not used PyCall much since 2015) 
`py"Foo.get_special_list]()"` returns a `Vector`,
and `py"Foo[:get_special_list]()"o` retrns a `PyObect`.

I guess `py"strings"` are the normal way to use PyCall now

**Comment by stevengj** [comment]
Yes, I guess it would be fine to be stricter about automatic conversions for dict and list.

**Comment by oxinabox** [comment]
As an aside, this would intrinsically just solve 
https://github.com/JuliaPy/PyCall.jl/issues/153
since if we don't juliaize anything that is not trivially juliaizable without loss of information,
there is no need for a custom iterator that avoids juliaizing things.
